controllers/events_controller.rb

```diff
@@ -0,0 +1,146 @@
+class EventsController < ApplicationController
+  before_action :require_organizer_or_admin, only: [ :destroy ]
+  before_action :require_organizer, only: [ :create, :update, :destroy ]
+  before_action :set_event, only: [ :show, :update, :destroy ]
+
+  def index
+    events = Event.includes(:location, :category, :event_discounts).all
+
+    events_with_banners = events.map do |event|
+      {
+        id: event.id,
+        title: event.title,
+        description: event.description,
+        start_datetime: event.start_datetime,
+        end_datetime: event.end_datetime,
+        location: event.location,
+        category: event.category,
+        event_discounts: event.event_discounts,
+        base_ticket_price: event.base_ticket_price,
+        capacity: event.capacity,
+        is_early_bird_active: event.is_early_bird_active,
+        is_amount_discount_active: event.is_amount_discount_active,
+        banner_url: event.banner.attached? ? url_for(event.banner) : nil
+      }
+    end
+
+    render json: events_with_banners
+  end
+
+
+  def show
+    render json: {
+      id: @event.id,
+      title: @event.title,
+      description: @event.description,
+      start_datetime: @event.start_datetime,
+      end_datetime: @event.end_datetime,
+      location: @event.location,
+      category: @event.category,
+      event_discounts: @event.event_discounts,
+      base_ticket_price: @event.base_ticket_price,
+      capacity: @event.capacity,
+      is_early_bird_active: @event.is_early_bird_active,
+      is_amount_discount_active: @event.is_amount_discount_active,
+      banner_url: @event.banner.attached? ? url_for(@event.banner) : nil
+    }
+  end
+
+
+  def create
```

Comment:
This is an action to create bookings. It should be written in bookings controller.